types/src/main/java/tech/pegasys/artemis/util/collections/ListImpl.java

```diff
@@ -65,9 +80,18 @@ public ValueType get(IndexType index) {
 
   @Override
   public boolean add(ValueType valueType) {
+    ensureCouldAddOne(valueType);
     return backedList.add(valueType);
   }
 
+  private void ensureCouldAddOne(ValueType value) {
```

Comment:
What if we replace these two methods with a general one? It could be `checkCapacity(IndexType newCapacity)`

Reply:
ok, done